homeassistant/components/tibber/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Adds config flow for Tibber integration."""
+import asyncio
+import logging
+
+import aiohttp
+import tibber
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_ACCESS_TOKEN): str})
+
+
+class TibberConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tibber integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+
+        if user_input is not None:
+            if self._async_current_entries():
```

Comment:
I think we can move this to the top since we should abort regardless of the form if there's already an entry, right?